source/minio-features/erasure-coding.rst

```diff
@@ -0,0 +1,277 @@
+   :width: 100%
+
+   * - Parity
+     - Total Storage
+     - Storage Ratio
+     - Minimum Drives for Read Operations
+     - Minimum Drives for Write Operations
+
+   * - ``EC: 4`` (Default)
+     - 12 Tebibytes
+     - 0.750
+     - 12
+     - 13
+
+   * - ``EC: 6``
+     - 10 Tebibytes
+     - 0.625
+     - 10
+     - 16
+
+   * - ``EC: 8``
+     - 8 Tebibytes
+     - 0.500
+     - 8
+     - 9
+
+- For more information on Erasure Sets, see :ref:`minio-ec-erasure-set`.
+
+- For more information on selecting Erasure Code Parity, see
+  :ref:`minio-ec-parity`
+
+- For more information on Erasure Code Object Healing, see
+  :ref:`minio-ec-object-healing`.
+
+.. _minio-ec-erasure-set:
+
+Erasure Sets
+------------
+
+An *Erasure Set* is a set of drives in a MinIO deployment that support
+Erasure Coding. MinIO evenly distributes object data and parity blocks among
+the drives in the Erasure Set. 
+
+MinIO calculates the number and size of *Erasure Sets* by dividing the total
+number of drives in the :ref:`Server Set <minio-intro-server-set>` into sets
+consisting of between 4 and 16 drives each. MinIO considers two factors when
+selecting the Erasure Set size:
+
+- The Greatest Common Divisor (GCD) of the total drives.
+
+- The number of :mc:`minio server` nodes in the Server Set.
+
+For an even number of nodes, MinIO uses the GCD to calculate the Erasure Set
+size and ensure the minimum number of Erasure Sets possible. For an odd number
+of nodes, MinIO selects a common denominator that results in an odd number of
+Erasure Sets to facilitate more uniform distribution of erasure set drives
+among nodes in the Server Set.
+
+For example, consider a Server Set consisting of 4 nodes with 8 drives each
+for a total of 32 drives. The GCD of 16 produces 2 Erasure Sets of 16 drives 
+each with uniform distribution of erasure set drives across all 4 nodes.
+
+Now consider a Server Set consisting of 5 nodes with 8 drives each for a total
+of 40 drives. Using the GCD, MinIO would create 4 erasure sets with 10 drives
+each. However, this distribution would result in uneven distribution with
+one node contributing more drives to the Erasure Sets than the others. 
+MinIO instead creates 5 erasure sets with 8 drives each to ensure uniform
+distribution of Erasure Set drives per Nodes.
+
+MinIO generally recommends maintaining an even number of nodes in a Server Set
+to facilitate simplified human calculation of the number and size of
+Erasure Sets in the Server Set.
+
+.. _minio-ec-parity:
+
+Erasure Code Parity (``EC:N``)
+------------------------------
+
+MinIO uses a Reed-Solomon algorithm to split objects into data and parity blocks
+based on the size of the Erasure Set. MinIO uses parity blocks to automatically
+heal damaged or missing data blocks when reconstructing an object. MinIO uses
+the ``EC:N`` notation to refer to the number of parity blocks (``N``) in the
+Erasure Set.
+
+When applications write a new object to the MinIO cluster, MinIO randomly
```

Comment:
Erasure set choice is not random per se, it is determined by a hash of the object name so an object with the same name will always land in the same erasure set.

Reply:
Tweaked